core/src/types/operator/operator.rs

```diff
@@ -1840,6 +1941,24 @@ impl Operator {
     /// # }
     /// ```
     ///
+    /// ## `version`
+    ///
+    /// Specify whether to list all object versions or not
+    ///
+    /// if `version` is not set, or is set to false, only the active files with the current version will be returned.
+    ///
+    /// when `version` is set to true and the backend service has `versioning` enabled, all object versions will be returned.
+    /// if `versioning` is not enabled, an error with kind [`ErrorKind::Unsupported`] will be returned.
```

Comment:
I will discuss this at https://github.com/apache/opendal/pull/5145. Generally, I don't want to declare it as part of our public API (or expected behavior).